freqtrade/plot/plotting.py

```diff
@@ -339,31 +342,34 @@ def analyse_and_plot_pairs(config: Dict[str, Any]):
 
     plot_elements = init_plotscript(config)
     trades = plot_elements['trades']
+    if trades is None:
+        raise ValueError('No trades to analyze')
```

Comment:
This is very limiting and i don't think it makes sense to handle this here

Not having trades should not fail the plot-dataframe - it'll simply plot no trades - which can be usefull to analyze a strategy while developing it (assume it creates no trades in backtesting - i would still like to see the indicators and maybe from seeing them i can determine why they're not generating sells).

if it does create errors, we should fix them where they happen - but when i tested this it did create log-outputs indicating that there are no trades.

Reply:
Given that I'm working on a strategy and want to visualize the results of a backtest
When I run the following in a Jupyter notebook:
```python
from freqtrade.plot.plotting import load_and_plot_trades
load_and_plot_trades(config)
```
I get:
```
2019-09-04 20:06:52,946 - freqtrade.data.history - INFO - Using local backtesting data ...
2019-09-04 20:06:57,340 - freqtrade.plot.plotting - INFO - analyse pair ETH/BTC
 AttributeError: 'NoneType' object has no attribute 'loc' 
---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
<ipython-input-20-0263847289e7> in <module>
      1 from freqtrade.plot.plotting import load_and_plot_trades
----> 2 load_and_plot_trades(config)

~/Documents/Repos/freqtrade/freqtrade/plot/plotting.py in load_and_plot_trades(config)
    356         logger.debug(f'pair: {pair}')
    357 
--> 358         trades_pair = trades.loc[trades['pair'] == pair]
    359         trades_pair = extract_trades_of_period(dataframe, trades_pair)
    360 

AttributeError: 'NoneType' object has no attribute 'loc'
```

This occurs because `init_plotscript()` generates `None` for the trades dataframe.

```
from freqtrade.plot.plotting import init_plotscript
plot_elements = init_plotscript(config)
display(plot_elements)

2019-09-04 20:11:41,978 - freqtrade.data.history - INFO - Using local backtesting data ...
{'tickers': {'ETH/BTC':                          date      open      high       low     close   volume
  0   2019-09-02 02:55:00+00:00  0.017519  0.017528  0.017495  0.017505  692.633
  1   2019-09-02 03:00:00+00:00  0.017506  0.017540  0.017505  0.017540  322.632
  ..                        ...       ...       ...       ...       ...      ...
  198 2019-09-02 19:25:00+00:00  0.017524  0.017533  0.017511  0.017514  499.837
  
  [199 rows x 6 columns]}, 'trades': None, 'pairs': ['ETH/BTC']}
```

My solution was to halt execution of `load_and_plot_trades` because generating a figure with no trades is not useful if I'm tweaking a strategy and visualizing results in a notebook.

Reply:
as mentioned above, especially early in strategy development it can be useful to just see the indicators (without having any buy signals yet), even without any trades happening.

I would handle this differently - however i wonder why this does not happen when running via cli - i guess you've got some parameters in config set "wrongly" so trades do not load from either DB nor backtesting - and a potential error is probably not shown in juptyer...

For jupyter - i don't think this is the ideal entry-point anyway - you should rather use the sample notebook to analyze the strategy - and you can call `generate_candlestick_graph()` from there on the dataframe directly and visualize the results in-line within the notebook (or open a new window to see it fullscreen).

That way you can avoid to set complicated config options which will vary deeply between iterations.